src/System-Support/SystemDictionary.class.st

```diff
@@ -455,6 +462,13 @@ SystemDictionary >> traitNames [
 			  ifAbsent: [ false ] ]
 ]
 
+{ #category : 'accessing' }
+SystemDictionary >> undeclaredRegistry [
+	"For now we are referencing a global variable of the default environment of Pharo. But in the future each environments should have their undeclared registry, else we have memory leaks of the environments."
```

Comment:
I think we do not need to back-port the undeclaredRegistry method, I think this is the reason why the bookstrap fails

Reply:
Thank you Marcus. Now the image is bootstrapped. 
The failing test seems unrelated to this PR 
<img width="1331" alt="image" src="https://github.com/pharo-project/pharo/assets/4822303/76a8e91a-f28f-4bcf-8558-8a96042ae196">